src/App.tsx

```diff
@@ -1,16 +1,26 @@
 import './App.scss';
+import { useState } from 'react';
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
 import moviesFromServer from './api/movies.json';
+import { Movie } from './types/Movie';
 
 export const App = () => {
+  const [newMovieList, setNewMovieList] = useState<Movie[]>(
+    [...moviesFromServer],
+  );
+
+  const setNewMovie = (movie:Movie) => {
+    setNewMovieList(current => [...current, movie]);
+  };
+
   return (
     <div className="page">
       <div className="page-content">
-        <MoviesList movies={moviesFromServer} />
+        <MoviesList movies={newMovieList} />
       </div>
       <div className="sidebar">
-        <NewMovie /* onAdd={(movie) => {}} */ />
+        <NewMovie onAdd={setNewMovie} />
```

Comment:
```suggestion
        <NewMovie onAdd={setNewMovie} key={count} />
```
Move counter logic here. When you'll increase count your component will be created from scratch with default values and no errors.